### PR TITLE
Set default value for jets w/o SV info in HI CSV tagger 

### PR DESCRIPTION
**Pull request by mandrenguyen**
#### PR description:

The heavy-ion version of the CSV tagger was introduced into 11_2 in #31660 
A peak is observed at a discriminator value of 0.30755
This was tracked down to change made during the review of #31660 which changed the way jets with no secondary vertex are identified.  
This PR reverts to the original logic, and these jets now take a value of -1, as they should. 
11_2_X backport is #32331 

#### PR validation:

Validated with wfs 140.5611 and 158.01

### CODE REVIEW

**Comment by slava77** [comment]
please test with workflow 140.5611

**Comment by perrotta** [comment]
please test workflow 140.5611

File: RecoBTag/Combined/src/HeavyIonCSVTagger.cc [hunk 1]
```diff
@@ -91,17 +91,17 @@ float HeavyIonCSVTagger::discriminator(const TagInfoHelper &tagInfo) const {
     //single value tagging var
     else {
       inputs[mva_var.name] = vars.get(mva_var.id, mva_var.default_value);
+      if (!vtxChecked) {
```
Comment (perrotta) [inline RecoBTag/Combined/src/HeavyIonCSVTagger.cc:94]: Can the following save a few cycles?

```suggestion
      if (noTrack) {
        if (mva_var.name == "TagVarCSV_vertexMass") {
          if (inputs[mva_var.name])<0.f)
            return -1.;         
          noTrack = false;
        }
      }
      
```

File: RecoBTag/Combined/src/HeavyIonCSVTagger.cc [hunk 2]
```diff
@@ -91,17 +88,14 @@ float HeavyIonCSVTagger::discriminator(const TagInfoHelper &tagInfo) const {
     //single value tagging var
     else {
       inputs[mva_var.name] = vars.get(mva_var.id, mva_var.default_value);
-      if (!vtxChecked) {
+      if (noTrack) {
         if (mva_var.name == "TagVarCSV_vertexMass") {
-          vtxMassVal = inputs[mva_var.name];
-          vtxChecked = true;
```
Comment (perrotta) [inline RecoBTag/Combined/src/HeavyIonCSVTagger.cc]: At this point you can set `noTrack = false;` (or give it a different name if `noTrack` does not fit any more with the updated functionality) and get rid of these checks in the following cycles, as you did before with  `vtxChecked = true;`

**Comment by perrotta** [comment]
@mandrenguyen, I think that the following image taken from the jenkins comparisons for wf 158.01 actually shows the effect of the bug (in the baseline) and the fix provided by this PR:
image

Out of curiosity: what does the value 0.30755 represent for the discriminator (i.e. why did it end up as being the "default" value)?

**Comment by mandrenguyen** [comment]
> @mandrenguyen, I think that the following image taken from the jenkins comparisons for wf 158.01 actually shows the effect of the bug (in the baseline) and the fix provided by this PR:
> image
> 
> Out of curiosity: what does the value 0.30755 represent for the discriminator (i.e. why did it end up as being the "default" value)?

Thanks for the comparison (and also the suggestions).  It would appear that if we send the xml file a negative mass value it will always return the same value for the discriminator.   I haven't dug into the file to understand why that happens, but at least it gives the same value every time, and not something random, which would have been harder to spot.

**Comment by perrotta** [comment]
@mandrenguyen (just in case you forgot it): if you apply here the last commit as you did for the 10_2 backport of this PR, I can launch the tests

**Comment by perrotta** [comment]
code-checks

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- It fixes the default value for the disciminator as intended
- Jenkins tests pass and show the effect of the fix in the HI miniAOD

**Comment by qliphy** [comment]
+1

**Pull request merged**